src/main/java/org/mskcc/cmo/messaging/impl/NatsGatewayImpl.java

```diff
@@ -0,0 +1,77 @@
+package org.mskcc.cmo.messaging.impl;
+
+import java.nio.charset.StandardCharsets;
+import java.util.HashMap;
+import java.util.Map;
+import org.mskcc.cmo.messaging.Gateway;
+import org.mskcc.cmo.messaging.MessageConsumer;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Component;
+import com.google.gson.Gson;
+import io.nats.client.Connection;
+import io.nats.streaming.Message;
+import io.nats.streaming.MessageHandler;
+import io.nats.streaming.StreamingConnection;
+import io.nats.streaming.Subscription;
+import io.nats.streaming.SubscriptionOptions;
+
+@Component
+public class NatsGatewayImpl implements Gateway {
+	
+	@Value("${nats.flush_duration:5}")
+	private Integer natsFlushDuration;
+    
+    @Autowired
+    private StreamingConnection stanConnection;
+
+    private Gson gson;
+    private Map<String, Subscription> subscribers;
+    private boolean initialized;
+
+
+    public NatsGatewayImpl(Connection natsConnection) {
```

Comment:
Is this constructor to support NATS core?  I would drop it unless we have a need for it.